src/Pact/Types/Command.hs

```diff
@@ -235,15 +236,13 @@ verifyUserSig msg UserSig{..} Signer{..} =
   where pubT = parseB16TextOnly _siPubKey
         sigT = parseB16TextOnly _usSig
         addrT = parseB16TextOnly <$> _siAddress
-        toScheme' = toScheme . fromMaybe ED25519
         isValidAddr addrM pubBS = case addrM of
           Nothing -> True
           Just (Left _) -> False
-          Just (Right givenAddr) ->
-            case formatPublicKeyBS (toScheme' _siScheme) (PubBS pubBS) of
-              Right expectAddr -> givenAddr == expectAddr
-              Left _           -> False
-
+          Just (Right givenAddr) -> case _siScheme of
```

Comment:
Why the snd `case _siSchema` and not `Just (Right givenAddr) -> pubBS == givenAddr` ?

Reply:
I should explain that in a comment.

The reason is: 
There are conceivably schemes (e.g. ETH) in which the address is not the same as the public key. I thought that by enumerating every case of `Scheme` here, we have a guarantee that every time someone adds a new scheme, they will encounter this place where they must specify how  addresses relate to pubkeys.

If I just enforce `pubBS == givenAddr`, the logic becomes wrong when someone adds a scheme with a different addr/pubkey relationship.

So the pattern-match here is meant for future-proofing, by letting the type checker tell us when we have to implement a new case. WDYT?

Reply:
I just had to delete a comment because this was confusing. I don't think the casing is worth the confusion. Finally I think we should reconsider removing ETH scheme, which presumably is different here

Reply:
Increasingly it seems like we're not going to support some other key scheme in Pact payloads, so address probably should be removed. No need to work on this now though.